src/Expensify.js

```diff
@@ -69,109 +62,51 @@ class Expensify extends PureComponent {
         // Initialize this client as being an active client
         ActiveClientManager.init();
 
-        this.removeLoadingState = this.removeLoadingState.bind(this);
-
         this.state = {
-            isLoading: true,
-            authToken: null,
             isOnyxMigrated: false,
         };
     }
 
     componentDidMount() {
-        // Run any Onyx schema migrations and then connect to Onyx
+        // Run any Onyx schema migrations and then continue loading the main app
         migrateOnyx()
             .then(() => {
                 this.setState({isOnyxMigrated: true});
-                Onyx.connect({
-                    key: ONYXKEYS.SESSION,
-                    callback: this.removeLoadingState,
-                });
             });
     }
 
-    componentDidUpdate(prevProps, prevState) {
-        if (this.state.accountID && this.state.accountID !== prevState.accountID) {
-            PushNotification.register(this.state.accountID);
+    componentDidUpdate(prevProps) {
+        const previousAccountID = lodashGet(prevProps, 'session.accountID', null);
+        const currentAccountID = lodashGet(this.props, 'session.accountID', null);
+        if (currentAccountID && (currentAccountID !== previousAccountID)) {
+            PushNotification.register(currentAccountID);
         }
     }
 
-    /**
-     * When the authToken is updated, the app should remove the loading state and handle the authToken
-     *
-     * @param {Object} session
-     * @param {String} session.authToken
-     */
-    removeLoadingState(session) {
-        this.setState({
-            authToken: session ? session.authToken : null,
-            accountID: session ? session.accountID : null,
-            isLoading: false,
-        });
-    }
-
     render() {
         // Until the authToken has been initialized from Onyx and the onyx migration is done, display a blank page
-        if (this.state.isLoading || !this.state.isOnyxMigrated) {
```

Comment:
Comment is no longer accurate, as the first condition has been removed.

Reply:
Nice catch